boa/src/builtins/string/mod.rs

```diff
@@ -486,7 +486,7 @@ impl String {
                     }
 
                     // Capture $1, $2, $3 etc
-                    if re.is_match(&result) {
+                    while re.is_match(&result) {
```

Comment:
I think this adds too much iteration, when we have the [`replace_all()`](https://docs.rs/regex/1.3.9/regex/struct.Regex.html#method.replace_all) method in `Regex`. I think we should still use an `if`, here, but in line 494, use `replace_all()` instead.

Reply:
I think we'd still need `while` though, `replace_all()` will only work for all ocurrences of `$x`. `$y` will need to be replaced in a different iteration, no?
I think the code block might need to be re-ordered, I think `$$1` will become `$1` and then be replaced by the capture group when it should just be `$1`.

Reply:
Hmmm I guess we need to iterate through the captures, but there should be only one check for the match, since the algorithm is not recursive. You would have the issue you mention